File: backend/app/schemas/file.py
"""文件上传相关请求/响应 Schema。"""

from datetime import datetime

from pydantic import BaseModel, Field


class FileUploadResponse(BaseModel):
    """文件上传成功响应。"""
    file_id: int
    original_name: str
    file_type: str
    mime_type: str
    file_size: int
    process_status: str
    created_at: datetime


class FileInfoResponse(BaseModel):
    """文件详细信息响应。"""
    file_id: int
    original_name: str
    file_type: str
    mime_type: str
    file_size: int
    process_status: str
    extracted_content: str | None = None
    extracted_metadata: dict | None = None
    image_description: str | None = None
    ocr_text: str | None = None
    error_message: str | None = None
    created_at: datetime
    processed_at: datetime | None = None


class FileListResponse(BaseModel):
    """文件列表响应。"""
    files: list[FileUploadResponse]
    total: int

File: frontend/src/api/file.ts
/**
 * 文件上传与管理 API。
 */

import api from '@/utils/api'

export interface FileUploadResponse {
  file_id: number
  original_name: string
  file_type: string
  mime_type: string
  file_size: number
  process_status: string
  created_at: string
}

export interface FileInfo extends FileUploadResponse {
  extracted_content?: string | null
  extracted_metadata?: Record<string, unknown> | null
  image_description?: string | null
  ocr_text?: string | null
  error_message?: string | null
  processed_at?: string | null
}

/** 上传文件 */
export function uploadFile(file: File, conversationId?: number) {
  const formData = new FormData()
  formData.append('file', file)

  const params = conversationId ? `?conversation_id=${conversationId}` : ''

  return api.post<FileUploadResponse>(`/files/upload${params}`, formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
    timeout: 60000, // 上传超时60秒
  })
}

/** 获取文件详情 */
export function getFileInfo(fileId: number) {
  return api.get<FileInfo>(`/files/${fileId}`)
}

/** 获取文件列表 */
export function listFiles(params?: { conversation_id?: number; page?: number; page_size?: number }) {
  return api.get('/files', { params })
}

/** 删除文件 */
export function deleteFile(fileId: number) {
  return api.delete(`/files/${fileId}`)
}
